### PR TITLE
extensions MENTZ TimedLeg and ExchangePoint

### PR DESCRIPTION
**Pull request by ue71603**
ServiceTrainNameExtension ==> added   ServiceGroup/TrainName
ServiceTrainNumberExtension ==> not done. TrainNumber exists in TimedLeg  ServiceGroup/TrainNumber
ServiceOperatorNameExtension ==> added ServiceGroup/OperatorName
TrackSectionExtension ==> not needed any more (is solved in #263)
ExchangePointWaitingTimeExtension ==> not done. WaitTime exists. Otherwise an explanation is needed, why this is not enough (ExchangePointsResultStructure/WaitTime)
ExchangePointPriorityExtension ==> added ExchangePointsResultStructure/Priority

### CODE REVIEW

File: OJP/OJP_Common.xsd [hunk 1]
```diff
@@ -398,4 +398,41 @@
...
+		<xs:annotation>
+			<xs:documentation>Elements about sustainability. Primarily for TRIPs and LEGs.</xs:documentation>
+		</xs:annotation>
+		<xs:sequence>
+			<xs:element name="EmissionCO2" minOccurs="0">
+				<xs:complexType>
+					<xs:sequence>
+						<xs:element name="GramPerPersonKm">
+							<xs:annotation>
+								<xs:documentation>Gram by person and by kilometer of CO2 emmisions.</xs:documentation>
```
Comment (skinkie) [inline OJP/OJP_Common.xsd:414]: SI would dictate kilogram.

File: OJP/OJP_Places.xsd [hunk 2]
```diff
@@ -321,7 +326,7 @@
 					<xs:documentation>Rough estimate of the travel duration from the specified refrence place to this exchange point.</xs:documentation>
 				</xs:annotation>
 			</xs:element>
-			<xs:element name="WaitDuration" type="xs:duration" minOccurs="0" default="PT0M">
+			<xs:element name="WaitDuration" type="xs:duration" default="PT0M" minOccurs="0">
```
Comment (skinkie) [inline OJP/OJP_Places.xsd:329]:
```suggestion
			<xs:element name="WaitDuration" type="xs:duration" minOccurs="0" default="PT0M" >
```

File: OJP/OJP_Places.xsd [hunk 3]
```diff
@@ -321,7 +326,7 @@
 					<xs:documentation>Rough estimate of the travel duration from the specified refrence place to this exchange point.</xs:documentation>
 				</xs:annotation>
 			</xs:element>
-			<xs:element name="WaitDuration" type="xs:duration" minOccurs="0" default="PT0M">
+			<xs:element name="WaitDuration" type="xs:duration" minOccurs="0" default="PT0M" >
```
Comment (ue71603) [inline OJP/OJP_Places.xsd:329]:
```suggestion
			<xs:element name="WaitDuration" type="xs:duration" minOccurs="0" default="PT0M">
```

**Comment by herlitze** [comment]
TrackSectionExtension: EFA-XML-Format is able to provide georeferences, since EFA-XML-Format is requested of active system, the active system needs to get this information from the passives systems via OJP. This is not needed in OJP standard, since it is a workaround for existing layouts.

File: OJP/OJP_Common.xsd (hunk 1 above)
Comment (ue71603) [reply OJP/OJP_Common.xsd:414]: True, but grap per personkm is the standard for CO2. So I would leave it. Can you agree @skinkie

File: OJP/OJP_FareSupport.xsd [hunk 4]
```diff
@@ -334,7 +334,14 @@
 			</xs:element>
 			<xs:element name="SaleUrl" type="WebLinkStructure" minOccurs="0" maxOccurs="unbounded">
 				<xs:annotation>
-					<xs:documentation>URL to buy the FareProduct online</xs:documentation>
+					<xs:documentation>URL to buy the FareProduct online. 
+					
+					Note: EU-Spirit:
```
Comment (Aurige) [inline OJP/OJP_FareSupport.xsd:339]: What do you mean here: is this specific to data produced for EU-Spirit ? It looks like a part of a profile put in the main standard document.

File: OJP/OJP_JourneySupport.xsd [hunk 5]
```diff
@@ -92,7 +92,17 @@
 					<xs:documentation>Contains the TrainNumber description from NeTEx. If several TrainNumber types exist, use the commercial number.</xs:documentation>
 				</xs:annotation>
 			</xs:element>
+			<xs:element name="TrainName" type="xs:normalizedString" minOccurs="0">
```
Comment (Aurige) [inline OJP/OJP_JourneySupport.xsd:95]: What is the name of a train ? Journey Pattern name ? Line name ? Train number ? Service Journey name ? or is it a branding attached to the train ?

File: OJP/OJP_JourneySupport.xsd [hunk 6]
```diff
@@ -467,7 +477,9 @@
 				</xs:element>
 				<xs:element name="TimeAllowance" type="xs:duration" minOccurs="0">
 					<xs:annotation>
-						<xs:documentation>Extra time needed before reaching/after leaving this place.</xs:documentation>
+						<xs:documentation>Extra time needed before reaching/after leaving this place.
+
```
Comment (Aurige) [inline OJP/OJP_JourneySupport.xsd:481]: I'm not sure what you mean by "Overloading the meaning in a responding system is discouraged."

File: OJP/OJP_Common.xsd (hunk 1 above)
Comment (ue71603) [reply OJP/OJP_Common.xsd:414]: http://www.aef.org.uk/downloads/Grams_CO2_transportmodesUK.pdf. So we do kilograms. 
Decision: kilogram

File: OJP/OJP_Common.xsd [hunk 7]
```diff
@@ -398,4 +398,41 @@
...
+							</xs:annotation>
+							<xs:simpleType>
+								<xs:restriction base="xs:decimal">
```
Comment (skinkie) [inline OJP/OJP_Common.xsd:417]:
```suggestion
								<xs:restriction base="xs:float">
```

File: OJP/OJP_Common.xsd [hunk 8]
```diff
@@ -398,4 +398,41 @@
...
+	</xs:annotation>
+	<xs:group name="SustainabilityGroup">
+		<xs:annotation>
+			<xs:documentation>Elements about sustainability. Primarily for TRIPs and LEGs.</xs:documentation>
+		</xs:annotation>
+		<xs:sequence>
+			<xs:element name="EmissionCO2" minOccurs="0">
+				<xs:complexType>
+					<xs:sequence>
+						<xs:element name="GramPerPersonKm">
```
Comment (skinkie) [inline OJP/OJP_Common.xsd:412]:
```suggestion
						<xs:element name="KilogramPerPersonKm">
```

File: OJP/OJP_Common.xsd (hunk 1 above)
Comment (skinkie) [inline OJP/OJP_Common.xsd:414]:
```suggestion
								<xs:documentation>Kilogram by person and by kilometer of CO2 emmisions.</xs:documentation>
```

File: OJP/OJP_FareSupport.xsd (hunk 4 above)
Comment (ue71603) [reply OJP/OJP_FareSupport.xsd:339]: => EU-Spirit behaviour will be moved to the EU-Spirit profile

Comment (ue71603) [reply OJP/OJP_FareSupport.xsd:339]: only remove line 339

File: OJP/OJP_JourneySupport.xsd (hunk 6 above)
Comment (ue71603) [reply OJP/OJP_JourneySupport.xsd:481]: Jona: expected time for the whole journey in their responsibility. For the long distance part. We need it somewhere.  Only Andreas Horst or Alexandre.  
Alexandre: We use it as a TimeOffset. n:m Trip Planning.

File: OJP/OJP_JourneySupport.xsd [hunk 9]
```diff
@@ -93,6 +93,11 @@
 				</xs:annotation>
 			</xs:element>
 			<xs:element ref="OperatorRef" minOccurs="0"/>
+			<xs:element name="OperatorName" type="xs:normalizedString" minOccurs="0">
```
Comment (skinkie) [inline OJP/OJP_JourneySupport.xsd:96]: Wasn't it decided to remove this?

File: OJP/OJP_Places.xsd [hunk 10]
```diff
@@ -334,7 +339,7 @@
 					<xs:documentation>Rough estimate of the travel duration from the specified refrence place to this exchange point.</xs:documentation>
 				</xs:annotation>
 			</xs:element>
-			<xs:element name="WaitDuration" type="xs:duration" minOccurs="0" default="PT0M">
+			<xs:element name="WaitDuration" type="xs:duration" default="PT0M" minOccurs="0">
```
Comment (skinkie) [inline OJP/OJP_Places.xsd:342]:
```suggestion
			<xs:element name="WaitDuration" type="xs:duration" minOccurs="0" default="PT0M">
```
Had the feeling I already created this suggestion.

File: OJP/OJP_Places.xsd [hunk 11]
```diff
@@ -349,6 +354,12 @@
 					<xs:documentation>List of transport modes that call at this place object. This list should only be filled in case of stop points or stop places – and only when explicitly requested.</xs:documentation>
 				</xs:annotation>
 			</xs:element>
+			<xs:element name="Priority" type="PercentType" minOccurs="0">
+				<xs:annotation>
+					<xs:documentation>The priority of the exchange point. 100 is the maximum.</xs:documentation>
+				</xs:annotation>
+				<!-- may be switched to type="Score" with "User Preferences TODO -->
```
Comment (skinkie) [inline OJP/OJP_Places.xsd:361]: Remove the comment?

File: OJP/OJP_Trips.xsd [hunk 12]
```diff
@@ -25,7 +25,15 @@
...
-					<xs:documentation>Ordered series of points where the journey must pass through. If more than one via point is given all of them must be obeyed - in the correct order. The server is allowed to replace a via stop by equivalent stops.</xs:documentation>
+					<xs:documentation>Ordered series of points where the journey must pass through. If more than one via point is given all of them must be obeyed - in the correct order. The server is allowed to replace a via stop by equivalent stops.
+					
+					Note: Systems may support only one.</xs:documentation>
+				</xs:annotation>
+			</xs:element>
+			<xs:element name="ViaSystem" type="siri:ParticipantRefStructure" minOccurs="0" maxOccurs="unbounded">
+				<xs:annotation>
+					<xs:documentation>With this parameter a distributing system is asked to build a trip using a given System to pass through. This helps in selecting Exchange Points for this trip. ViaSystem is also used in sequence.
+					Note: Systems may support only one.</xs:documentation>
```
Comment (skinkie) [inline OJP/OJP_Trips.xsd:36]:
```suggestion
					<xs:documentation>With this parameter a distributing system is asked to build a trip using a given System to pass through. This helps in selecting Exchange Points for this trip. ViaSystem is also used in sequence. Note: Systems may support only one.</xs:documentation>
```

File: OJP/OJP_Trips.xsd [hunk 13]
```diff
@@ -865,12 +875,38 @@
...
-			<xs:enumeration value="eachDestination"/>
+			<xs:enumeration value="anyPoint">
+				<xs:annotation>
+					<xs:documentation>if a solution for any one of multiple origin/destination points is sufficient. Some implementation will return only one origin-destination pair even for multiple results with this setting. Others will provide multiple. </xs:documentation>
+				</xs:annotation>
+			</xs:enumeration>
+			<xs:enumeration value="eachOrigin">
+				<xs:annotation>
+					<xs:documentation>a distinct solution for each of the 
+origin points has to be found.</xs:documentation>
```
Comment (skinkie) [inline OJP/OJP_Trips.xsd:892]:
```suggestion
					<xs:documentation>a distinct solution for each of the origin points has to be found.</xs:documentation>
```

File: OJP/OJP_Places.xsd (hunk 11 above)
Comment (ue71603) [reply OJP/OJP_Places.xsd]: done

File: OJP/OJP_Common.xsd [hunk 14]
```diff
@@ -420,4 +420,41 @@
...
+								<xs:restriction base="xs:float">
+									<xs:minInclusive value="0"/>
+								</xs:restriction>
+							</xs:simpleType>
+						</xs:element>
+						<xs:element name="ConfidenceLevel" minOccurs="0">
+							<xs:annotation>
+								<xs:documentation>Confidence level of the emission value in percent.</xs:documentation>
+							</xs:annotation>
+							<xs:simpleType>
```
Comment (Aurige) [inline OJP/OJP_Common.xsd:448]: I think that a percentage type was recently introduced ..... it may be used here

File: OJP/OJP_Trips.xsd [hunk 15]
```diff
@@ -414,6 +422,7 @@
 					<xs:documentation>Geographic embedding of this leg.</xs:documentation>
 				</xs:annotation>
 			</xs:element>
+			<xs:group ref="SustainabilityGroup" minOccurs="0"/>
```
Comment (Aurige) [inline OJP/OJP_Trips.xsd:425]: Why don't you put it at TripLegStructure level so it is available for all ?

File: OJP/OJP_Common.xsd (hunk 14 above)
Comment (ue71603) [reply OJP/OJP_Common.xsd:448]: done

File: OJP/OJP_Trips.xsd (hunk 15 above)
Comment (ue71603) [reply OJP/OJP_Trips.xsd:425]: I moved it. TransferLegs hopefully are quite sustainable. But it does not hurt.

**Review by skinkie — APPROVED** [review]
Personally: I think this is... "too woke".

**Review by Aurige — APPROVED** [review]
Multiple topics in the same PR but Ok

**Review by trurlurl — COMMENTED** [review]
How is the ConfidenceLevel defined, what does 20% or 50% mean?

**Review by trurlurl — APPROVED** [review]

File: OJP/OJP_Common.xsd [hunk 16]
```diff
@@ -425,4 +425,35 @@
...
+							<xs:annotation>
+								<xs:documentation>Kilogram by person and by kilometer of CO2 emmisions.</xs:documentation>
+							</xs:annotation>
+							<xs:simpleType>
+								<xs:restriction base="xs:float">
+									<xs:minInclusive value="0"/>
+								</xs:restriction>
+							</xs:simpleType>
+						</xs:element>
+						<xs:element name="ConfidenceLevel" type="Percentage" minOccurs="0">
```
[the remark this answers is not in the document]
Comment (ue71603) [reply OJP/OJP_Common.xsd:449]: Even with all its problems, I would keep it that way. It is from EU Spirit.

File: OJP/OJP_Trips.xsd [hunk 17]
```diff
@@ -933,12 +943,37 @@
...
+				</xs:annotation>
+			</xs:enumeration>
+			<xs:enumeration value="eachOrigin">
+				<xs:annotation>
+					<xs:documentation>a distinct solution for each of the origin points has to be found.</xs:documentation>
+				</xs:annotation>
+			</xs:enumeration>
+			<xs:enumeration value="eachDestination">
+				<xs:annotation>
+					<xs:documentation>a distinct solution for each of the destination points has to be found.</xs:documentation>
```
Comment (skinkie) [inline OJP/OJP_Trips.xsd:964]:
```suggestion
					<xs:documentation>A distinct solution for each of the destination points has to be found.</xs:documentation>
```

File: OJP/OJP_Trips.xsd [hunk 18]
```diff
@@ -933,12 +943,37 @@
...
-			<xs:enumeration value="eachOrigin"/>
-			<xs:enumeration value="eachDestination"/>
+			<xs:enumeration value="anyPoint">
+				<xs:annotation>
+					<xs:documentation>if a solution for any one of multiple origin/destination points is sufficient. Some implementation will return only one origin-destination pair even for multiple results with this setting. Others will provide multiple. </xs:documentation>
+				</xs:annotation>
+			</xs:enumeration>
+			<xs:enumeration value="eachOrigin">
+				<xs:annotation>
+					<xs:documentation>a distinct solution for each of the origin points has to be found.</xs:documentation>
```
Comment (skinkie) [inline OJP/OJP_Trips.xsd:959]:
```suggestion
					<xs:documentation>A distinct solution for each of the origin points has to be found.</xs:documentation>
```

File: OJP/OJP_Trips.xsd [hunk 19]
```diff
@@ -933,12 +943,37 @@
...
+			Note: MultiPointType trumps the number of results. If the algorithm thinks it necessary to provide more results, it should.
+			</xs:documentation>
 		</xs:annotation>
 		<xs:restriction base="xs:string">
-			<xs:enumeration value="anyPoint"/>
-			<xs:enumeration value="eachOrigin"/>
-			<xs:enumeration value="eachDestination"/>
+			<xs:enumeration value="anyPoint">
+				<xs:annotation>
+					<xs:documentation>if a solution for any one of multiple origin/destination points is sufficient. Some implementation will return only one origin-destination pair even for multiple results with this setting. Others will provide multiple. </xs:documentation>
```
Comment (skinkie) [inline OJP/OJP_Trips.xsd:954]:
```suggestion
					<xs:documentation>If a solution for any one of multiple origin/destination points is sufficient. Some implementation will return only one origin-destination pair even for multiple results with this setting. Others will provide multiple. </xs:documentation>
```

**Review by skinkie — DISMISSED** [review]

**Review by trurlurl — APPROVED** [review]

**Review by AndreasAtSBB — APPROVED** [review]
Typo fixes and extending doc in xsd-files and in HTML file.

**Review by sgrossberndt — CHANGES_REQUESTED** [review]
This pull request is not up to date and still contains the mentioned typo "emmissions"

File: OJP/OJP_Common.xsd [hunk 20]
```diff
@@ -420,4 +420,35 @@
...
+		<xs:annotation>
+			<xs:documentation>Elements about sustainability. Primarily for TRIPs and LEGs.</xs:documentation>
+		</xs:annotation>
+		<xs:sequence>
+			<xs:element name="EmissionCO2" minOccurs="0">
+				<xs:complexType>
+					<xs:sequence>
+						<xs:element name="KilogramPerPersonKm">
+							<xs:annotation>
+								<xs:documentation>Kilogram by person and by kilometer of CO2 emmisions.</xs:documentation>
```
[the remark this answers is not in the document]
Comment (ue71603) [reply OJP/OJP_Common.xsd]: changed

**Review by skinkie — APPROVED** [review]

**Review by AndreasAtSBB — APPROVED** [review]
Changes OK.

File: OJP/OJP_Places.xsd [hunk 21]
```diff
@@ -364,7 +369,7 @@
 					<xs:documentation>Rough estimate of the travel duration from the specified reference place to this exchange point.</xs:documentation>
 				</xs:annotation>
 			</xs:element>
-			<xs:element name="WaitDuration" type="xs:duration" default="PT0M" minOccurs="0">
+			<xs:element name="WaitDuration" type="xs:duration" minOccurs="0" default="PT0M">
```
Comment (sgrossberndt) [inline OJP/OJP_Places.xsd:372]: I really dislike this merge frenzy without approval from several people.
This merge reverted https://github.com/VDVde/OJP/pull/286 again.
Why?

**Pull request merged**